tests/trans/test_adjoint.F90

```diff
@@ -322,44 +323,44 @@ SUBROUTINE SCALPRODSP(PSP1,PSP2,PSC)
 
 END SUBROUTINE SCALPRODSP
 
-SUBROUTINE SCALPRODGP(PGP1,PGP2,PSC)
-
-!Scalar product in gridpoint space
-
-REAL(KIND=JPRB) :: PGP1(:,:,:),PGP2(:,:,:)
-REAL(KIND=JPRB) :: PSC
-
-INTEGER(KIND=JPIM) :: JLEV,JKGLO,JROF,IEND,IBL,IOFF,IROF,JGL,JL
-REAL(KIND=JPRB) :: ZGP(NPROMA,NFLEVG,NGPBLKS),ZGPG(NGPTOTG,NFLEVG),ZWTG(NGPTOTG,1),ZWT(NPROMA,1,NGPBLKS)
-
-PSC = 0.0_JPRB
-ZGP(:,:,:) = 0.0_JPRB
-
-IROF=0
-DO JGL=1,NDGL
-  DO JL=1,NLOEN(JGL)
-    IROF=IROF+1
-    ZWTG(IROF,1) = 1.0_JPRB
-  ENDDO
-ENDDO
-CALL DIST_GRID(PGPG=ZWTG,KFDISTG=1,KFROM=ITO,PGP=ZWT,KPROMA=NPROMA)
-
-DO JLEV=1,NFLEV
-  DO JKGLO=1,NGPTOT,NPROMA
-    IEND = MIN(NPROMA,NGPTOT-JKGLO+1)
-    IOFF = JKGLO-1
-    IBL  = (JKGLO-1)/NPROMA+1
-    DO JROF=1,IEND
-      ZGP(JROF,JLEV,IBL) = PGP1(JROF,JLEV,IBL)*PGP2(JROF,JLEV,IBL)*&
-       &ZWT(JROF,1,IBL)
-    ENDDO
-  ENDDO
-ENDDO
-
-CALL GATH_GRID(PGPG=ZGPG,KFGATHG=NFLEVG,KTO=ITO,PGP=ZGP,KPROMA=NPROMA)
-
-PSC = SUM(ZGPG)
-
-END SUBROUTINE SCALPRODGP
+!SUBROUTINE SCALPRODGP(PGP1,PGP2,PSC)
+!
+!!Scalar product in gridpoint space
+!
+!REAL(KIND=JPRB) :: PGP1(:,:,:),PGP2(:,:,:)
+!REAL(KIND=JPRB) :: PSC
+!
+!INTEGER(KIND=JPIM) :: JLEV,JKGLO,JROF,IEND,IBL,IOFF,IROF,JGL,JL
+!REAL(KIND=JPRB) :: ZGP(NPROMA,NFLEVG,NGPBLKS),ZGPG(NGPTOTG,NFLEVG),ZWTG(NGPTOTG,1),ZWT(NPROMA,1,NGPBLKS)
+!
+!PSC = 0.0_JPRB
+!ZGP(:,:,:) = 0.0_JPRB
+!
+!IROF=0
+!DO JGL=1,NDGL
+!  DO JL=1,NLOEN(JGL)
+!    IROF=IROF+1
+!    ZWTG(IROF,1) = 1.0_JPRB
+!  ENDDO
+!ENDDO
+!CALL DIST_GRID(PGPG=ZWTG,KFDISTG=1,KFROM=ITO,PGP=ZWT,KPROMA=NPROMA)
+!
+!DO JLEV=1,NFLEV
+!  DO JKGLO=1,NGPTOT,NPROMA
+!    IEND = MIN(NPROMA,NGPTOT-JKGLO+1)
+!    IOFF = JKGLO-1
+!    IBL  = (JKGLO-1)/NPROMA+1
+!    DO JROF=1,IEND
+!      ZGP(JROF,JLEV,IBL) = PGP1(JROF,JLEV,IBL)*PGP2(JROF,JLEV,IBL)*&
+!       &ZWT(JROF,1,IBL)
+!    ENDDO
+!  ENDDO
+!ENDDO
+!
+!CALL GATH_GRID(PGPG=ZGPG,KFGATHG=NFLEVG,KTO=ITO,PGP=ZGP,KPROMA=NPROMA)
+!
+!PSC = SUM(ZGPG)
+!
+!END SUBROUTINE SCALPRODGP
```

Comment:
What's the justification for commenting this out?

Reply:
Because this routine is reported as unused.

Reply:
It may still be worthwhile to compile it anyhow, serving as example?

Reply:
Commenting out code should be a temporary thing IMO.

Reply:
Awaiting decision on this one, then.

Reply:
I looked at the code, it seems straightforward. Best to delete dead code altogether, please.

Reply:
done